src/Analyzers/CSharp/Tests/NamespaceSync/NamespaceSyncTests.cs

```diff
@@ -0,0 +1,565 @@
+{{
+    class RefClass
+    {{
+        private Class1Alias c1;
+    }}
+}}";
+
+            var fixed1 =
+@"namespace A.B.C
+{
+    class Class1
+    {
+    }
+}";
+
+            var fixed2 =
+@"
+using System;
+using Class1Alias = A.B.C.Class1;
+
+namespace Foo
+{
+    class RefClass
+    {
+        private Class1Alias c1;
+    }
+}";
+
+            var originalSources = new[]
+            {
+                (Path.Combine(folder, "Class1.cs"), code1),
+                ("Class2.cs", code2)
+            };
+
+            var fixedSources = new[]
+            {
+                (Path.Combine(folder, "Class1.cs"), fixed1),
+                ("Class2.cs", fixed2)
+            };
+
+            await RunTestAsync(originalSources, fixedSources);
+        }
+
+        [Fact]
+        public async Task FixAll()
+        {
+            var declaredNamespace = "Bar.Baz";
+
+            var folder1 = CreateFolderPath("A", "B", "C");
+            var fixedNamespace1 = "A.B.C";
+
+            var folder2 = CreateFolderPath("Second", "Folder", "Path");
+            var fixedNamespace2 = "Second.Folder.Path";
+
+            var folder3 = CreateFolderPath("Third", "Folder", "Path");
+            var fixedNamespace3 = "Third.Folder.Path";
+
+            var code1 =
+$@"namespace [|{declaredNamespace}|]
+{{
+    class Class1
+    {{
+        Class2 C2 {{ get; }}
+        Class3 C3 {{ get; }}
+    }}
+}}";
+
+            var fixed1 =
+$@"using {fixedNamespace2};
+using {fixedNamespace3};
+
+namespace {fixedNamespace1}
+{{
+    class Class1
+    {{
+        Class2 C2 {{ get; }}
+        Class3 C3 {{ get; }}
+    }}
+}}";
+
+            var code2 =
+$@"namespace [|{declaredNamespace}|]
+{{
+    class Class2
+    {{
+        Class1 C1 {{ get; }}
+        Class3 C3 {{ get; }}
+    }}
+}}";
+
+            var fixed2 =
+$@"using {fixedNamespace1};
+using {fixedNamespace3};
+
+namespace {fixedNamespace2}
+{{
+    class Class2
+    {{
+        Class1 C1 {{ get; }}
+        Class3 C3 {{ get; }}
+    }}
+}}";
+
+            var code3 =
+$@"namespace [|{declaredNamespace}|]
+{{
+    class Class3
+    {{
+        Class1 C1 {{ get; }}
+        Class2 C2 {{ get; }}
+    }}
+}}";
+
+            var fixed3 =
+$@"using {fixedNamespace1};
+using {fixedNamespace2};
+
+namespace {fixedNamespace3}
+{{
+    class Class3
+    {{
+        Class1 C1 {{ get; }}
+        Class2 C2 {{ get; }}
+    }}
+}}";
+
+            var sources = new[]
+            {
+                (Path.Combine(folder1, "Class1.cs"), code1),
+                (Path.Combine(folder2, "Class2.cs"), code2),
+                (Path.Combine(folder3, "Class3.cs"), code3),
+            };
+
+            var fixedSources = new[]
+            {
+                (Path.Combine(folder1, "Class1.cs"), fixed1),
+                (Path.Combine(folder2, "Class2.cs"), fixed2),
+                (Path.Combine(folder3, "Class3.cs"), fixed3),
+            };
+
+            await RunTestAsync(sources, fixedSources);
+        }
+    }
+}
```

Comment:
Is there tests with `RootNamespace` defined?

Reply:
No, and honestly the support should be removed because C# doesn't have root namespace.

Reply:
> C# doesn't have root namespace.

This is not quite true. While C# doesn't have the concept of root namespace that is understood by compiler (like in VB, all namespaces declared in source are implicitly under the root namespace), it does have something called "Default namespace", defined with the same "rootnamespace" property, which, as far as I know, only exists for tooling support (e.g. new class in VS will be created under default namespace). People have been using default namespace for sync feature (it's supported by our refactoring and resharper). For example, if the default namespace is `A.B.C` and the document is under subfolder "D", then the expect namespace declaration in the doc would be `A.B.C.D`.

![image](https://user-images.githubusercontent.com/788783/105422658-f7107500-5bf8-11eb-9d7f-ec7cff7fd963.png)

Reply:
That's kind of crazy that default namespace gets exposed the same way. I didn't know that...